src/Service/SchemaGenerator.php

```diff
@@ -0,0 +1,259 @@
+<?php
+
+namespace SilverStripe\MFA\Service;
+
+use SilverStripe\Control\HTTPRequest;
+use SilverStripe\Core\Extensible;
+use SilverStripe\Core\Injector\Injectable;
+use SilverStripe\MFA\Exception\MemberNotFoundException;
+use SilverStripe\MFA\Extension\MemberExtension;
+use SilverStripe\MFA\Store\StoreInterface;
+use SilverStripe\ORM\FieldType\DBDate;
+use SilverStripe\Security\Member;
+use SilverStripe\Security\Security;
+use SilverStripe\SiteConfig\SiteConfig;
+
+/**
+ * Generates a multi-factor authentication frontend app schema from the given request
+ */
+class SchemaGenerator
+{
+    use Extensible;
+    use Injectable;
+
+    /**
+     * @var HTTPRequest
+     */
+    protected $request;
+
+    /**
+     * @var StoreInterface
+     */
+    protected $store;
+
+    /**
+     * @param HTTPRequest $request
+     * @param StoreInterface $store
+     */
+    public function __construct(HTTPRequest $request, StoreInterface $store)
+    {
+        $this->setRequest($request);
+        $this->setStore($store);
+    }
+
+    /**
+     * Gets the schema data for the multi factor authentication app, using the current Member as context
+     *
+     * @return array
+     */
+    public function getSchema()
+    {
+        $registeredMethods = $this->getRegisteredMethods();
+
+        // Skip registration details if the user has already registered this method
+        $exclude = array_column($registeredMethods, 'urlSegment');
+
+        $schema = [
+            'registeredMethods' => $registeredMethods,
+            'availableMethods' => $this->getAvailableMethods($exclude),
+            'defaultMethod' => $this->getDefaultMethod(),
+            'canSkip' => $this->canSkipMFA(),
+            'shouldRedirect' => $this->shouldRedirectToMFA(),
+        ];
+
+        $this->extend('updateSchema', $schema);
+
+        return $schema;
+    }
+
+    /**
+     * @return Member|MemberExtension
+     * @throws MemberNotFoundException
+     */
+    public function getMember()
+    {
+        $member = $this->store->getMember() ?: Security::getCurrentUser();
```

Comment:
I think I prefer having `$member` as an argument for `getSchema` then this class doesn't have to have the concern of resolving the member.